playbooks/roles/cert-manager/tasks/main.yml

```diff
@@ -0,0 +1,49 @@
+---
+
+- name: Download pip3
+  apt:
+    name: python3-pip
+
+- name: Download pipenv
+  pip:
+    executable: pip3
+    name: pipenv
+
+- name: Download cert-manager
+  git:
+    repo: "https://github.com/open-craft/cert-manager"
+    dest: "{{ cert_manager_path }}"
+
+- name: Install cert-manager requirements
+  shell: pipenv install
+  args:
+    chdir: "{{ cert_manager_path }}"
+
+- name: Copy over cert-manager script and service files
+  template:
+    src: "{{ item.src }}"
+    dest: "{{ item.dest }}"
+    mode: "{{ item.mode }}"
+  with_items:
+    - src: "cert-manager.sh"
+      dest: "/usr/local/sbin/cert-manager.sh"
+      mode: "0755"
+    - src: "cert-manager.service"
+      dest: "/lib/systemd/system/cert-manager.service"
+      mode: "0644"
+
+- name: Enable and start the cert-manager service
+  systemd:
+    daemon_reload: yes
+    enabled: yes
+    state: restarted
+    name: "cert-manager"
+
+- name: Open HTTP and HTTPS port on the firewall
+  ufw:
+    rule: allow
+    port: "{{ item }}"
+    proto: tcp
+  with_items:
+    - 80
+    - 443
```

Comment:
What are these ports being opened for here?

Reply:
80 for ACME challenges, and not sure what 443 is for, I guess it's "standard" :p